app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt

```diff
@@ -325,21 +326,31 @@ class RecipeRepository(private val db: FirebaseFirestore) {
    * Fetches recipes that contain any of the given ingredient IDs and maps them to Recipe objects.
    *
    * @param ingredientIds The list of ingredient IDs to match against.
+   * @param lastRecipe The last recipe fetched in the previous query. If null, fetches the first
    * @param onSuccess A callback function invoked with the list of fetched recipes.
    * @param onFailure A callback function invoked on failure to fetch the recipes, with an
    *   exception.
    */
   private fun fetchAndMapRecipes(
       ingredientIds: List<String>,
-      onSuccess: (List<Recipe>) -> Unit,
+      lastRecipe: DocumentSnapshot?,
+      onSuccess: (List<Recipe>, DocumentSnapshot?) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
     if (ingredientIds.isEmpty()) {
       onFailure(IllegalArgumentException())
       return
     }
-    db.collection(collectionPath)
-        .whereArrayContainsAny("ingredientIds", ingredientIds)
+    // Create a query to fetch the top rated recipes
+    var queryRef =
+        db.collection(collectionPath).whereArrayContainsAny("ingredientIds", ingredientIds).limit(6)
```

Comment:
No, you cannot do that, here you limit the recipes to 6. Hence the ranking will happen only on those 6 recipes and not all the recipe that match criteria. Either increase this limit or do a filtering on the highest ranked recipes first.
The best way will be to utilize SQL firebase or BigQuerry but it was decided that local filtering will work for now

Reply:
Hmmm I see, so where do you think can I add this `.limit()` call ? (I did these changes to be able to add the Load more button)

Reply:
The process is done in "one go" 
You can't really pause it and get a coherent result. 
(Would be the case if it was a not local suggestion)

Hence, there is no load more feature. The output is a complete list, for the UI you could use a LazyList and it should display only the requested element.

Reply:
I wanted to implement the Load more button functionality, but I think I will just display all fetched recipes on after the other.